packages/j/juce/xmake.lua

```diff
@@ -0,0 +1,237 @@
+            readonly = false,
+            links = {
+                iphoneos = {"OpenGLES"},
+                macosx = {"OpenGL"},
+                linux = {"hl"},
+                mingw = {"opengl32"}
+            },
+            deps = {"juce_gui_extra"}
+        },
+        juce_osc = {
+            default = false,
+            readonly = false,
+            deps = {"juce_events"}
+        },
+        juce_product_unlocking = {
+            default = false,
+            readonly = false,
+            deps = {"juce_cryptography"}
+        },
+        juce_video = {
+            default = false,
+            readonly = false,
+            links = {
+                iphoneos = {"AVKit", "AVFoundation", "CoreMedia"},
+                macosx = {"AVKit", "AVFoundation", "CoreMedia"},
+            },
+            deps = {"juce_cryptography"}
+        }
+    }
+
+    for modulename, options in table.orderpairs(modules) do
+        add_configs(modulename, {description = format("Enable %s module", modulename:gsub("_", " ")), default = options.default, type = "boolean", readonly = options.readonly})
+    end
+
+    add_configs("utf", {description = "Set the character encoding type", default = "8", values = {"8", "16", "32"}})
+
+    on_install(function (package)
+        package:add("defines", "JUCE_GLOBAL_MODULE_SETTINGS_INCLUDED")
+        local file = format([[
+            add_rules("mode.debug", "mode.release")
+            target("juce")
+                set_kind("$(kind)")
+                set_languages("cxx17")
+                add_defines("JUCE_GLOBAL_MODULE_SETTINGS_INCLUDED", {public = true})
+                add_defines("JUCE_STRING_UTF_TYPE=8", {public = true})
+                if is_kind("shared") then
+                    add_defines("JUCE_DLL_BUILD")
+                end
+        ]], package:config("utf"))
+        for modulename, options in pairs(modules) do
+            local links = options.links or {}
+            local sysLinks = '"' .. table.concat(links[os.host()] or {}, '", "') .. '"'
+            for _, link in ipairs(links[os.host()] or {}) do
+                package:add("links", link)
+            end
+            if package:config(modulename) then
+                file = file .. format([[
+                    do
+                        local module = "%s"
+                        add_files("modules/" .. module .. "/" .. module .. ".cpp")
+                        add_includedirs("modules/", { public = true })
+                        add_headerfiles("modules/(" .. module .. "/" .. module .. ".h)")
+                        for _, dir in ipairs(os.dirs("modules/" .. module .. "/**")) do
+                            add_includedirs(dir, { public = true })
+                            add_headerfiles("modules/(" .. dir:gsub("modules\\", "") .. "/*.h)")
+                        end
+                        add_syslinks(%s)
+                    end
+                ]], modulename, sysLinks)
+            end

```

Comment:
please add xmake.lua file to port/xmake.lua, and config all options in xmake.lua

see https://github.com/xmake-io/xmake-repo/blob/dev/packages/i/imgui/port/xmake.lua